### ISSUE TITLE
Issue with building the React JS modules for mainline

### ISSUE THREAD

**Issue by dwayne-hart** [DFO-Ocean-Navigator/Ocean-Data-Map-Project#856]
```
(navigator) vagrant@ubuntu-focal:~/Ocean-Data-Map-Project/oceannavigator/frontend$ yarn install
yarn install v1.22.10
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.3.2: The platform "linux" is incompatible with this module.
info "fsevents@2.3.2" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
Done in 28.12s.
(navigator) vagrant@ubuntu-focal:~/Ocean-Data-Map-Project/oceannavigator/frontend$ yarn build
yarn run v1.22.10
$ NODE_ENV=production node --stack_size=1000000 node_modules/.bin/webpack --progress -p
Hash: 83eb17fbe8f47c58a8d4                                                                
Version: webpack 3.6.0
Time: 21890ms
                              Asset       Size  Chunks                    Chunk Names
  /glyphicons-halflings-regular.eot    20.1 kB          [emitted]         
/glyphicons-halflings-regular.woff2      18 kB          [emitted]         
 /glyphicons-halflings-regular.woff    23.4 kB          [emitted]         
  /glyphicons-halflings-regular.ttf    45.4 kB          [emitted]         
  /glyphicons-halflings-regular.svg     109 kB          [emitted]         
       /ui-icons_444444_256x240.png    3.76 kB          [emitted]         
       /ui-icons_555555_256x240.png    3.76 kB          [emitted]         
       /ui-icons_ffffff_256x240.png    3.76 kB          [emitted]         
       /ui-icons_777620_256x240.png    3.76 kB          [emitted]         
       /ui-icons_cc0000_256x240.png    3.76 kB          [emitted]         
       /ui-icons_777777_256x240.png    3.76 kB          [emitted]         
           /fontawesome-webfont.eot     166 kB          [emitted]         
         /fontawesome-webfont.woff2    77.2 kB          [emitted]         
          /fontawesome-webfont.woff      98 kB          [emitted]         
           /fontawesome-webfont.ttf     166 kB          [emitted]         
           /fontawesome-webfont.svg     444 kB          [emitted]  [big]  
                       /spinner.gif    41.3 kB          [emitted]         
                             /x.png  153 bytes          [emitted]         
                            /I0.svg  131 bytes          [emitted]         
                            /I1.svg  747 bytes          [emitted]         
                            /I2.svg  747 bytes          [emitted]         
                            /I3.svg  747 bytes          [emitted]         
                            /I4.svg  745 bytes          [emitted]         
                            /I5.svg  747 bytes          [emitted]         
                            /I6.svg  747 bytes          [emitted]         
                            /I7.svg  747 bytes          [emitted]         
                            /I8.svg  747 bytes          [emitted]         
                            /I9.svg  745 bytes          [emitted]         
                  /sad-computer.png     2.1 kB          [emitted]         
                    /bar_loader.gif    10.8 kB          [emitted]         
                  oceannavigator.js    7.57 MB       0  [emitted]  [big]  main
                 oceannavigator.css     343 kB       0  [emitted]  [big]  main
                         index.html    1.87 kB          [emitted]         
  [20] ./src/i18n.js 828 bytes {0} [built]
  [75] (webpack)/buildin/global.js 509 bytes {0} [built]
 [626] ./src/currentLanguage.js 93 bytes {0} [built]
 [640] multi babel-polyfill ./src/index.jsx 40 bytes {0} [built]
 [843] ./src/index.jsx 2.37 kB {0} [built]
[1052] ../translations/fr.json 15.6 kB {0} [built]
[1053] ../translations/en.json 3.29 kB {0} [built]
[1055] ./src/images/x.png 52 bytes {0} [built]
[1056] ./src/images/s111/I0.svg 53 bytes {0} [built]
[1438] ./src/remote/OceanNavigator.js 633 bytes {0} [built]
[1457] ./src/images/bar_loader.gif 61 bytes {0} [built]
[1463] ./src/stylesheets/utils/bootstrap.css 41 bytes {0} [built]
[1464] ./src/stylesheets/main.scss 41 bytes {0} [built]
[1512] ./node_modules/css-loader!./src/stylesheets/utils/bootstrap.css 120 kB [built]
[1513] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/stylesheets/main.scss 13.6 kB [built]
    + 1499 hidden modules

ERROR in oceannavigator.js from UglifyJs
Unexpected token: keyword (const) [oceannavigator.js:60314,9]
Child html-webpack-plugin for "index.html":
     1 asset
       [0] ./node_modules/html-webpack-plugin/lib/loader.js!./src/index.ejs 2.14 kB {0} [built]
       [2] (webpack)/buildin/global.js 509 bytes {0} [built]
       [3] (webpack)/buildin/module.js 517 bytes {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/ol/ol.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/jquery-ui-month-picker/src/MonthPicker.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/rc-slider/assets/index.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/react-datepicker/dist/react-datepicker.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/react-bootstrap-toggle/dist/bootstrap2-toggle.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/jquery-ui/themes/base/datepicker.css:
       2 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!src/stylesheets/utils/bootstrap.css:
       [0] ./node_modules/css-loader!./src/stylesheets/utils/bootstrap.css 120 kB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/bootstrap/dist/css/bootstrap.css:
     5 assets
       7 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/jquery-ui/themes/base/theme.css:
     6 assets
       8 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/jquery-ui/themes/base/base.css:
       21 modules
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!src/stylesheets/main.scss:
       [0] ./node_modules/css-loader!./node_modules/sass-loader/lib/loader.js!./src/stylesheets/main.scss 13.6 kB {0} [built]
        + 1 hidden module
Child extract-text-webpack-plugin node_modules/extract-text-webpack-plugin/dist node_modules/css-loader/index.js!node_modules/sass-loader/lib/loader.js!node_modules/font-awesome/scss/font-awesome.scss:
     5 assets
       8 modules
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by htmlboss** [comment]
yeah sorry I meant to mention this earlier. a workaround would be `yarn dev`, wait for the build to finish and then do a `CTRL + C` to stop the watching process, but the debug JS build will still work fine.

There's a really nasty dependency problem I need to untangle now. We're running a 5 year old JS configuration which I need to incrementally updated../.

**Comment by htmlboss** [comment]
I'll put up a PR this afternoon to tweak the `yarn build` so that it builds in debug mode for now.

**Comment by dwayne-hart** [comment]
The error pertains to the following code snipet in [...]/public/oceannavigator.js.

```
function sortKeysBy(obj, reference) {
    const result = {};
    for (const key in reference) {
        if (obj[key] !== undefined) {
            result[key] = obj[key];
        }
    }
    for (const key in obj) {
        if (result[key] === undefined) {
            result[key] = obj[key];
        }
    }
    return result;
}
```

Some people have used UglifyJS with Butternut or replaced UglifyJS with gulp-uglify-es. The following is a stackoverflow [page](https://stackoverflow.com/questions/47439067/uglifyjs-throws-unexpected-token-keyword-const-with-node-modules) providing discussion about possible work arounds.

**Comment by dwayne-hart** [comment]
It is interesting that using the **yarn dev** command built the node_modules without issue and I was able to get my build environment running the new code you introduced for the velocity quiver plot.

**Comment by dwayne-hart** [comment]
Is the quiver arrows pointing in the right direction along the Labrador coast?

![image](https://user-images.githubusercontent.com/44408505/124765987-141fd280-df11-11eb-9321-6ac42a6085f2.png)

**Comment by htmlboss** [comment]
No idea. I just copy-pasted the bearing code from the CSV generation into the calculation layer https://github.com/DFO-Ocean-Navigator/Ocean-Data-Map-Project/blob/b003f786eed835c1cbb71da72fea0120adfece73/data/calculated_parser/functions.py#L60

That bearing code has been around for about 4 years now but it's never been double-checked.

**Comment by htmlboss** [comment]
> It is interesting that using the **yarn dev** command built the node_modules without issue and I was able to get my build environment running the new code you introduced for the velocity quiver plot.

Something to do with production builds increasing the strictness of compiling JS code in one of our dependencies 🤷

**Comment by dwayne-hart** [comment]
If I were a betting person. I would say that the **yarn dev** brings in the debugging symbols allowing devs to perform a trace through a debugging app.

**Comment by jmunroe** [comment]
I agree that looks completely flipped from what we normally would expect for the Labrador current.  I'll look at some other models and find out if this correct or not.

> Is the quiver arrows pointing in the right direction along the Labrador coast?
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/44408505/124765987-141fd280-df11-11eb-9321-6ac42a6085f2.png)
> 
>

**Comment by jmunroe** [comment]
Yes. It appears we have our velocity vectors flipped. Here's what another tool views this same data as:

![Web capture_7-7-2021_10265_meteocentre com](https://user-images.githubusercontent.com/6181563/124777088-f2711c00-df0d-11eb-9894-5d56df43d84e.jpeg)

[source](https://meteocentre.com/plus/?srs=EPSG%3A4326&bbox=-69.78226693069307,48.074708910891076,-41.788666930693054,65.97952633663365&service=https%3A%2F%2Fgeo.weather.gc.ca%2Fgeomet-beta%2F%3Flang%3Den&layer=OCEAN.GIOPS.2D_UU2W%24image%2Fpng%24true%24SEA_CURARROW_0MTO100M_SURFACE%241%240&selected=0&dims=time$2021-07-17T00:00:00Z,reference_time$2021-07-07T00:00:00Z&baselayers=naturalearth2$countryborders)`

for the same GIOPS data as viewed in a different viewer.

**Comment by dwayne-hart** [comment]
@htmlboss is there a way to use the east_vector component and nort_vector_component brought in to ensure that the magnitudes were calculated correctly?

> "magwatervel": { "name": "Water Velocity", "envtype": "ocean", "unit": "m/s", "scale": [0, 3], "equation": "magnitude(vozocrtx, vomecrty)",  "dims": ["time", "depth", "latitude", "longitude"], "east_vector_component": "vozocrtx", "north_vector_component": "vomecrty" },

**Comment by htmlboss** [comment]
Crash with running `yarn build` resolved in #865
